include/h2o.h

```diff
@@ -2038,10 +2047,18 @@ void h2o_status_register_configurator(h2o_globalconf_t *conf);
 
 /* lib/handler/headers_util.c */
 
+struct st_yoml_t;
+struct st_h2o_headers_add_arg_t {
+    struct st_yoml_t *node;
+    h2o_iovec_t *name;
+    h2o_iovec_t value;
+};
+typedef struct st_h2o_headers_add_arg_t h2o_headers_add_arg_t;
```

Comment:
I get the hassle, but I am not sure if it is a good idea to expose `st_yoml_t *` here.

libh2o does not depend on libyoml, therefore it would be safer to contain the use of `yoml_t` within the source files of lib/handler/configurator/* and src/* (and therefore the definition and use of `st_h2o_headers_add_arg_t` in lib/handler/configurator/headers_util.c).